iceoryx_posh/test/moduletests/test_roudi_iceoryx_roudi_app.cpp

```diff
@@ -0,0 +1,178 @@
+
+    auto cmdLineArgs = cmdLineParser.parse(numberOfArgs, args);
+
+    IceoryxRoudiApp_Child roudi(cmdLineArgs.value(), iox::RouDiConfig_t().setDefaults());
+
+    EXPECT_TRUE(roudi.getVariableRun());
+    EXPECT_THAT(roudi.getLogLevel(), Eq(iox::log::LogLevel::kWarn));
+    EXPECT_THAT(roudi.getMonitoringMode(), Eq(roudi::MonitoringMode::ON));
+}
+
+TEST_F(IceoryxRoudiApp_test, CreateTwoRoudiAppIsSuccessfull)
+{
+    constexpr uint8_t numberOfArgs{1U};
+    char* args[numberOfArgs];
+    char appName[] = "./foo";
+    args[0] = &appName[0];
+
+    auto cmdLineArgs = cmdLineParser.parse(numberOfArgs, args);
+
+    IceoryxRoudiApp_Child roudi(cmdLineArgs.value(), iox::RouDiConfig_t().setDefaults());
+
+    IceoryxRoudiApp_Child roudiTest(cmdLineArgs.value(), iox::RouDiConfig_t().setDefaults());
+
+    EXPECT_TRUE(roudiTest.getVariableRun());
+}
+
+TEST_F(IceoryxRoudiApp_test, VerifyRunMethodWithFalseConditionReturnExitSuccess)
+{
+    constexpr uint8_t numberOfArgs{1U};
+    char* args[numberOfArgs];
+    char appName[] = "./foo";
+    args[0] = &appName[0];
+
+    auto cmdLineArgs = cmdLineParser.parse(numberOfArgs, args);
+
+    IceoryxRoudiApp_Child roudi(cmdLineArgs.value(), iox::RouDiConfig_t().setDefaults());
+
+    roudi.setVariableRun(false);
+
+    uint8_t result = roudi.callRun();
+
+    EXPECT_EQ(result, EXIT_SUCCESS);
+}
+
+TEST_F(IceoryxRoudiApp_test, ConstructorCalledWithArgUniqueIdTwoTimesReturnError)
+{
+    constexpr uint8_t numberOfArgs{3U};
+    char* args[numberOfArgs];
+    char appName[] = "./foo";
+    char option[] = "--unique-roudi-id";
+    char value[] = "4242";
+    args[0] = &appName[0];
+    args[1] = &option[0];
+    args[2] = &value[0];
+
+    auto cmdLineArgs = cmdLineParser.parse(numberOfArgs, args);
+
+    iox::cxx::optional<iox::Error> detectedError;
+    auto errorHandlerGuard = iox::ErrorHandler::SetTemporaryErrorHandler(
+        [&detectedError](const iox::Error error, const std::function<void()>, const iox::ErrorLevel errorLevel) {
+            detectedError.emplace(error);
+            EXPECT_THAT(errorLevel, Eq(iox::ErrorLevel::MODERATE));
+        });
+
+    IceoryxRoudiApp_Child roudi(cmdLineArgs.value(), iox::RouDiConfig_t().setDefaults());
+
+    IceoryxRoudiApp_Child roudiTest(cmdLineArgs.value(), iox::RouDiConfig_t().setDefaults());
+
+    ASSERT_THAT(detectedError.has_value(), Eq(true));
+    EXPECT_THAT(detectedError.value(), Eq(iox::Error::kPOPO__TYPED_UNIQUE_ID_ROUDI_HAS_ALREADY_DEFINED_UNIQUE_ID));
+}
+
+TEST_F(IceoryxRoudiApp_test, ConstructorCalledWithArgVersionSetRunVariableToFalse)
+{
+    constexpr uint8_t numberOfArgs{2U};
+    char* args[numberOfArgs];
+    char appName[] = "./foo";
+    char option[] = "-v";
+    args[0] = &appName[0];
+    args[1] = &option[0];
+
+    auto cmdLineArgs = cmdLineParser.parse(numberOfArgs, args);
+
+    IceoryxRoudiApp_Child roudi(cmdLineArgs.value(), iox::RouDiConfig_t().setDefaults());
+
+    EXPECT_FALSE(roudi.getVariableRun());
+}
+
```

Comment:
What about tests when memory segments/mempool config is empty?
Can the sighandler setting be testedl, e.g. by checking the semaphore count?

Reply:
Is it possible to provide a roudi config empty ?

Reply:
Added the tests for roudi config.